refactor: wire SecurityManager & providers (#10)

hi-malik opened the pull request:
### Description
This PR finishes the refactoring of our authentication flow by centralizing provider selection in `SecurityManager`, delegating real DB checks to `UserService`, and loading SSO domains from config (Issue #10): 

### Backend Changes
- **DatabaseSecurityProvider**  
  • Now injects `UserService`  
  • Returns success/failure based on real credentials  
- **EntraExternalIdSSOProvider**  
  • Placeholder implementation that returns “not implemented” until full SSO flow is added  
- **AuthenticationFilter**  
  • Reads `sso.enabled-domains` (comma-separated) from `application.properties`  
  • Registers the SSO provider for each domain at startup  
  • Routes both `/login` and all other protected requests through `SecurityManager`  
- **Config**  
  • Added `sso.enabled-domains=sso-company.com,partner.org` to `application.properties`

### Frontend Changes
No code changes here—the Angular login page and auth service remain unchanged. Requests still send `X-Email`/`X-Password` headers to the same endpoints.

### How to Test
1. **Start services**  
   • Backend:  
     ```bash
     cd backend
     mvn spring-boot:run
     ```  
   • Frontend:  
     ```bash
     cd frontend
     npm start
     ```  
2. **DB-backed login**  
   • In the Angular app login form (or via curl):  
     ```bash
     curl -i -X POST http://localhost:8080/login \
       -H "X-Email: user@internal.com" \
       -H "X-Password: password"
     ```  
   • Expect HTTP 200 and a JSON response with the email.  
3. **SSO-domain login (should fail “not implemented”)**  
   • Use an SSO domain configured in `sso.enabled-domains` (e.g. `alice@sso-company.com`).  
   • Expect HTTP 401 and `{ "error": "Invalid credentials" }`.  

### Preview
![2025-07-3117-06-10-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/adf0b529-4d5b-4198-bf9c-a6d40b7f682b)

Closes #10

mondalsantanu commented on backend/src/main/java/com/intsof/samples/entra/filter/AuthenticationFilter.java:
```diff
@@ -43,51 +46,65 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
+                res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
                 res.setContentType("application/json");
-                res.getWriter().write("{\"error\": \"Invalid credentials\"}");
+                res.getWriter().write("{\"error\": \"Email header missing\"}");
```
avoid hard coding values, keep all constant in an interface e.g.
public interface ApplicationConstants {
    String INVALID_CREDENTIAL = "Invalid credentials";

mondalsantanu commented on backend/src/main/java/com/intsof/samples/entra/filter/AuthenticationFilter.java:
```diff
@@ -25,6 +25,9 @@ public class AuthenticationFilter implements Filter {
     @Autowired
     private UserService userService;
 
+    @org.springframework.beans.factory.annotation.Value("${sso.enabled-domains:}")
```
@org.springframework.beans.factory.annotation.Value("${sso.enabled-domains}")
Use @Value

hi-malik commented:
### Refactor SecurityManager / Providers

- Extracted all magic strings (HTTP headers, JSON keys & error messages, content-type) into a new `ApplicationConstants` interface.
- Simplified `AuthenticationFilter`:
  - Removed all inline comments and duplicated `setContentType` calls.
  - Consolidated header reads (`X-Email` / `X-Password`) and JSON response formatting.
  - Moved login logic into a dedicated `handleLogin(...)` method.
  - Switched SSO domain injection to `@Value("${sso.enabled-domains}") List<String>` and stream-registered providers.
- Updated `DatabaseSecurityProvider` to reference `ApplicationConstants.ERROR_INVALID_CREDENTIALS` instead of hard-coded text.
- Cleaned up unused fields (e.g. `frontEndLoginUrl`) and removed leftover TODOs/comments.
- No functional changes; behavior remains the same for DB auth and deferred Entra SSO setup.

mondalsantanu commented:
Ready to merge the changes
